src/olympia/landfill/serializers.py

```diff
@@ -28,10 +35,53 @@
 from olympia.users.models import UserProfile
 from olympia.devhub.tasks import create_version_for_upload
 
+log = olympia.core.logger.getLogger('z.users')
+
 
 class GenerateAddonsSerializer(serializers.Serializer):
     count = serializers.IntegerField(default=10)
 
+    def __init__(self):
+        self.user = None
+        self.fxa_email = os.environ['FXA_EMAIL']
+        self.fxa_password = os.environ['FXA_PASSWORD']
+        self.fxa_id = None
+        self._create_fxa_user()
+        self._create_addon_user()
+
+    def _create_fxa_user(self):
+        """Create fxa user for logging in."""
+        fxa_client = Client(ENVIRONMENT_URLS['stable']['authentication'])
+        account = TestEmailAccount(email=self.fxa_email)
+        password = self.fxa_password
+        FxAccount = collections.namedtuple('FxAccount', 'email password')
+        fxa_account = FxAccount(email=account.email, password=password)
+        session = fxa_client.create_account(fxa_account.email,
+                                            fxa_account.password)
+        account.fetch()
+        message = account.wait_for_email(
+            lambda m: 'x-verify-code' in m['headers'] and
+            session.uid == m['headers']['x-uid']
+        )
+        session.verify_email_code(message['headers']['x-verify-code'])
+        self.fxa_id = session.uid
+        log.debug("fxa account created: {}".format(fxa_account))
```

Comment:
I really hate side-effects in classes like that (we have them everywhere, and they're always a pain to debug).

Let's pass `fxa_password` and `fxa_email` as arguments to `_create_fxa_user` and `_create_addon_user` as well as pass the `fxa_id` to `_create_addon_user` by returning it from `_create_fxa_user`.

```python
def __init__(self):
    fxa_id = _create_fxa_user(username, password)
    self.user = _create_addon_user(fxa_id)
```

We still have that side-effect of setting `self.user` but I'd say that's okay and is how this class generally works.